### ISSUE TITLE
more documents required!

### ISSUE THREAD

**Issue by yield22** [amznlabs/amazon-dsstne#6]
Thanks for open source! It looks like a great package! But currently it seems hard to follow with the limited documents, in particular:

1. the data set ml20m-all is not in a normal user-item-rating format, and thus I am not sure what should be the input of the network, and what exactly problems the network can solve (rating prediction, ranking? clearly it is more about recommendation rather than image classification I can see).
2. the network structures currently available file seem all to be autoencoder, it would be nice to see some documents describing these networks for the specific recommendation problem, and other network structures that the package supports (or probably used in Amazon).

**Comment by rgeorgej** [comment]
Thanks @yield22 for the feedback. Answers to your questions are below

1 . You can use this for Rating prediction,Ranking or use it for regression. It will boil down to how you formulate the problem for the Neural Network to learn.
2.  You are absolutely correct that the current file is for auto encoder.  We will add more documentation describing these networks. As we add more features we will be adding more network structures that the package supports

Do let us know your feedback and concerns as we value it a lot

**Comment by elanmart** [comment]
Hi, I'd also love to see a more complete documentation, for example about the data format expected.

Readme states it should look like this:
`Example1    Feature1:Feature2:Feature3`
but what does it mean? Feature1 is and index? A value? Both? In what order?

**Comment by rgeorgej** [comment]
Analog input as Signals is supported in the DSSTNE Engine but not exposed in the wrapper .  Only Digital Signals are supported.  

Feature1 is a featureName 

Order is not important as the entire example is taken and input layer activated 

We will be exposing on how to do analog signals soon

**Comment by olihb** [comment]
Hi @rgeorgej, can you discribe the sample ml20m-all dataset? It's a bit difficult to map the content of this file to the movie lens dataset (they contain three files). Thanks!

**Comment by rgeorgej** [comment]
ml-20m has 4 files links.csv,movies.csv,ratings.csv  and tags.csv. For the example here we took the ratings.csv file and flattened out to to userId and their  rating time stamps. 
1,2,3.5,1112486027
1,29,3.5,1112484676
1,32,3.5,1112484819
1,47,3.5,1112484727
1,50,3.5,1112484580
1,112,3.5,1094785740
1,151,4.0,1094785734
1,223,4.0,1112485573
1,253,4.0,1112484940....

was converted to  
1      2,1112486027:29,1112484676:32,1112484819:47,1112484727:50,1112484580:112,1094785740:151,1094785734:223,1112485573:253,........


Do let us know if you need any more clarification
